Tests/External/Plugins/ASCompletion.Tests/Completion/ASCompleteTests.cs

```diff
@@ -606,6 +606,162 @@ public ComaExpression Haxe(string text)
+                    yield return new TestCaseData("{}+}").     Returns("{}").      SetName(prefix + "Close } to overwrite }");
+                    yield return new TestCaseData("\"\"+\"").  Returns("\"\"").    SetName(prefix + "Close \" to overwrite \"");
+                    yield return new TestCaseData("\"\\\"+\"").Returns("\"\\\"\"").SetName(prefix + "Close \" escaped should not overwrite \"");
+                    yield return new TestCaseData("''+'").     Returns("''").      SetName(prefix + "Close ' to overwrite '");
+                    yield return new TestCaseData("'\\'+'").   Returns("'\\''").   SetName(prefix + "Close ' escaped should not overwrite '");
+                    yield return new TestCaseData("<>+>").     Returns("<>").      SetName(prefix + "Close > to overwrite >");
+                    #endregion
+                    
+                    #region Test Delete
+                    yield return new TestCaseData("(-)").  Returns("").SetName(prefix + "Delete ( to delete )");
+                    yield return new TestCaseData("[-]").  Returns("").SetName(prefix + "Delete [ to delete ]");
+                    yield return new TestCaseData("{-}").  Returns("").SetName(prefix + "Delete { to delete }");
+                    yield return new TestCaseData("\"-\"").Returns("").SetName(prefix + "Delete \" to delete \"");
+                    yield return new TestCaseData("'-'").  Returns("").SetName(prefix + "Delete ' to delete '");
+                    yield return new TestCaseData("<->").  Returns("").SetName(prefix + "Delete < to delete >");
+                    #endregion
+
+                    #region Test Around Strings
+                    yield return new TestCaseData("\"\"(+").     Returns("\"\"()").   SetName(prefix + "Open ( after 'String' before 'Default'");
+                    yield return new TestCaseData("\"\"(+\"\""). Returns("\"\"(\"\"").SetName(prefix + "Open ( after 'String' before 'String'");
+                    yield return new TestCaseData("\"\")+)").    Returns("\"\")").    SetName(prefix + "Close ) after 'String' to overwrite )");
+                    yield return new TestCaseData("\"\")+)\"\"").Returns("\"\")\"\"").SetName(prefix + "Close ) after 'String' before 'String' to overwrite )");
+                    yield return new TestCaseData("\"\"(-)").    Returns("\"\"").     SetName(prefix + "Delete ( after 'String' to delete )");
+                    yield return new TestCaseData("\"\"(-)\"\"").Returns("\"\"\"\""). SetName(prefix + "Delete ( after 'String' before 'String' to delete )");
+
+                    yield return new TestCaseData("\"(+\""). Returns("\"(\""). SetName(prefix + "Open ( inside a string");
+                    yield return new TestCaseData("\")+)\"").Returns("\"))\"").SetName(prefix + "Close ) inside a string");
+                    yield return new TestCaseData("\"(-)\"").Returns("\")\""). SetName(prefix + "Delete ( inside a string");
```

Comment:
You could make a many sets of test cases, instead of using regions
like as https://github.com/SlavaRa/fdplugin-postfix-code-completion/blob/master/PostfixCodeCompletionTests/Completion/PostfixGeneratorTests.cs#L668